_posts/2016-04-25-exrm-release.md

```diff
@@ -0,0 +1,218 @@
+
+You have dependencies (direct/transitive) which are not in :applications!
+The following apps should to be added to :applications in mix.exs:
+
+        phoenix_live_reload => phoenix_live_reload is missing from favorite_products
+
+Continue anyway? Your release may not work as expected if these dependencies are required! [Yn]: Y
+
+
+You have dependencies (direct/transitive) which are not in :applications!
+The following apps should to be added to :applications in mix.exs:
+
+        phoenix_live_reload => phoenix_live_reload is missing from extensions_manager
+
+Continue anyway? Your release may not work as expected if these dependencies are required! [Yn]: Y
+```
+
+In the end we can see:
+
+```bash
+==> The release for user_app-0.0.1 is ready!
+==> You can boot a console running your release with `$ rel/user_app/bin/user_app console`
+```
+
+Let's try accessing it. running with no args gives us the output:
+
+```bash
+Usage: user_app {start|start_boot <file>|foreground|stop|restart|reboot|ping|rpc <m> <f> [<a>]|console|console_clean|console_boot <file>|attach|remote_console|upgrade|escript|command <m> <f> <args>}
+```
+
+We need to start the application to access the user_app and nectar, so let' do that. We are using foreground to see the console output, it can be run as a background service. See here
+
+```bash
+user_app foreground
+```
+
+and we see
+
+{% raw %}
+```bash
+[info] Application user_app exited: UserApp.start(:normal, []) returned an error: shutdown: failed to start child: UserApp.Endpoint
+    ** (EXIT) shutdown: failed to start child: Phoenix.CodeReloader.Server
+        ** (EXIT) an exception was raised:
+            ** (UndefinedFunctionError) undefined function Mix.Project.config/0 (module Mix.Project is not available)
+                Mix.Project.config()
+                (phoenix) lib/phoenix/code_reloader/server.ex:32: Phoenix.CodeReloader.Server.init/1
+                (stdlib) gen_server.erl:328: :gen_server.init_it/6
+                (stdlib) proc_lib.erl:240: :proc_lib.init_p_do_apply/3
+{"Kernel pid terminated",application_controller,"{application_start_failure,user_app,{{shutdown,{failed_to_start_child,'Elixir.UserApp.Endpoint',{shutdown,{failed_to_start_child,'Elixir.Phoenix.CodeReloader.Server',{undef,[{'Elixir.Mix.Project',config,[],[]},{'Elixir.Phoenix.CodeReloader.Server',init,1,[{file,\"lib/phoenix/code_reloader/server.ex\"},{line,32}]},{gen_server,init_it,6,[{file,\"gen_server.erl\"},{line,328}]},{proc_lib,init_p_do_apply,3,[{file,\"proc_lib.erl\"},{line,240}]}]}}}}},{'Elixir.UserApp',start,[normal,[]]}}}"}
+[1]    3000 user-defined signal 2  ./user_app foreground
+```
+
+It seems to be unable to start the endpoint because the code reloader is missing. We can add to the applications, but since it has no reason for belonging in the release. We will comment out the config for now in dev.exs and set code_reload: false in config.exs, doing that and building a new release.
```

Comment:
We can add **it** to the applications but since it has no reason for belonging in the release**, we** will comment out the config for now...